pkg/collector/corechecks/servicediscovery/apm/detect.go

```diff
@@ -134,48 +138,49 @@ func pythonDetector(args []string, envs map[string]string) Instrumentation {
 	return None
 }
 
+// isNodeInstrumented parses the provided `os.File` trying to find an
+// entry for APM NodeJS instrumentation. Returns true if finding such
+// an entry, false otherwise.
+func isNodeInstrumented(f *os.File) bool {
+	const readLimit = 16 * 1024 // Read 16KiB max
+
+	limitReader := io.LimitReader(f, readLimit)
```

Comment:
Maybe we do additionally want to use a stat check like `canSafelyParse` here similar to the service name code, to ensure the file has a size and is not a device file or something before reading? A read(2) on a device file could hang even for a small size.

Reply:
I added a stat check to not try to parse non-regular files.